packages/block-library/src/media-text/edit.js

```diff
@@ -229,81 +234,123 @@ function MediaTextEdit( { attributes, isSelected, setAttributes } ) {
 	};
 
 	const mediaTextGeneralSettings = (
-		<PanelBody title={ __( 'Settings' ) }>
-			<RangeControl
-				__nextHasNoMarginBottom
-				__next40pxDefaultSize
-				label={ __( 'Media width' ) }
-				value={ temporaryMediaWidth || mediaWidth }
-				onChange={ commitWidthChange }
-				min={ WIDTH_CONSTRAINT_PERCENTAGE }
-				max={ 100 - WIDTH_CONSTRAINT_PERCENTAGE }
-			/>
-			<ToggleControl
-				__nextHasNoMarginBottom
+		<ToolsPanel
+			label={ __( 'Settings' ) }
+			resetAll={ resetAll }
+			dropdownMenuProps={ TOOLSPANEL_DROPDOWNMENU_PROPS }
+		>
+			{ mediaUrl && (
+				<ToolsPanelItem
+					label={ __( 'Media width' ) }
+					isShownByDefault
+					hasValue={ () => !! mediaWidth }
+					onDeselect={ () => setAttributes( { mediaWidth: 50 } ) }
+				>
+					<RangeControl
+						__nextHasNoMarginBottom
+						label={ __( 'Media width' ) }
+						value={ temporaryMediaWidth || mediaWidth }
+						onChange={ commitWidthChange }
+						min={ WIDTH_CONSTRAINT_PERCENTAGE }
+						max={ 100 - WIDTH_CONSTRAINT_PERCENTAGE }
+					/>
+				</ToolsPanelItem>
+			) }
+			<ToolsPanelItem
 				label={ __( 'Stack on mobile' ) }
-				checked={ isStackedOnMobile }
-				onChange={ () =>
-					setAttributes( {
-						isStackedOnMobile: ! isStackedOnMobile,
-					} )
+				isShownByDefault
+				hasValue={ () => !! isStackedOnMobile }
```

Comment:
```suggestion
				hasValue={ () => ! isStackedOnMobile }
```

I think it is necessary to detect that it has been changed from the default value of `true`.